### ISSUE TITLE
Enable the ability to cycle through Parameters of Native Effects and Generators

### ISSUE THREAD

**Issue by chinakov** [audacity/audacity#8768]
### Description

If the User has opened a dialog window of an effect in Destructive or Real Time effect mode, he should be able to tab to each parameter of said effect via 'Tab' when the User has navigated through the effect header buttons in Realtime or Destructive Effect dialog windows. 

Here is an example of the order via native 'reverb' effect: 

![Image](https://github.com/user-attachments/assets/1312b171-5e1b-4bbf-b736-5c302daecd8f)

Here's the full order of tabbing starting with enable/disable button. 

![Image](https://github.com/user-attachments/assets/198f2dc7-972d-455a-8c32-8d92eb7b8d2e)

**Comment by saintmatthieu** [comment]
Estimate: 1 week

**Comment by Eism** [comment]
We typically divide dialogs into panels, with each panel assigned to a section - one for the entire dialog. For example, in the dialog from the issue, we can divide the dialog into two or three panels: a top panel, a left panel, and a right panel / or top and bottom panels. Navigating between these panels is accomplished by pressing Tab, and within each panel, navigation is accomplished by using arrow keys.
So we don’t navigate through elements using Tab - that’s the old approach. But there were ideas to somehow change the approach for dialogs - to allow tab navigation there, while keeping the current approach in the main window. However, this is debatable and requires more detailed research. Therefore, for consistency in the use of approaches, it would be more correct to split the dialog into panels.

**Comment by ianalarina** [comment]
this is required for screen readers support which is marked as P0. bumping priority here too

**Comment by DavidBailes** [comment]
> So we don’t navigate through elements using Tab - that’s the old approach. But there were ideas to somehow change the approach for dialogs - to allow tab navigation there, while keeping the current approach in the main window. However, this is debatable and requires more detailed research. Therefore, for consistency in the use of approaches, it would be more correct to split the dialog into panels.

@Eism . I think that using standard Tab navigation within group boxes in dialogs would be much better. I have opened a discussion here:
https://github.com/audacity/audacity/discussions/10256

**Comment by teetow** [comment]
Design decision:

- We stay with the current system: Tab moves between "panels," arrows navigate between controls within each panel. We'll be open to revising this system in the future, but it's what we've built and used elsewhere in the app. It'll be on the implementer to "slice" popups into sensible panels.
- Supporting screen readers goes beyond just tab navigation. This is a fairly broad subject, but stuff like setting proper roles for all (custom) controls, grouping things correctly (including group names) matter a lot here.

**Comment by dozzzzer** [comment]
@grliszas14 

1. The track header (or whichever UI element was put in focus last) is put in focus after the effect's dialog is closed:

https://github.com/user-attachments/assets/a0390cd8-5ba5-4f5f-acc4-95881694f5a9

2. The time component does not look fully outlined when in focus:

<img width="512" height="290" alt="Image" src="https://github.com/user-attachments/assets/c9775959-4e43-4b76-bf8e-84f36bc090e6" />

**Comment by dozzzzer** [comment]
Tested
